src/data/dhis2/TrackedEntityInstances.ts

```diff
@@ -0,0 +1,217 @@
+        organisationUnits: orgUnits,
+        ouMode: relationshipsOuFilter,
+    });
+
+    // Avoid 414-uri-too-large by spliting orgUnit in chunks
+    const orgUnitsList = _.chunk(orgUnits, 250);
+
+    // Get TEIs for the first page:
+    const apiTeis: TrackedEntityInstanceApi[] = [];
+
+    for (const orgUnits of orgUnitsList) {
+        // Limit response size by requesting paginated TEIs
+        for (let page = 1; ; page++) {
+            const { pager, trackedEntityInstances } = await getTeisFromApi({
+                api,
+                program,
+                orgUnits,
+                page,
+                pageSize,
+                enrollmentStartDate,
+                enrollmentEndDate,
+                ouMode: relationshipsOuFilter,
+            });
+            apiTeis.push(...trackedEntityInstances);
+            if (pager.pageCount <= page) break;
+        }
+    }
+
+    return apiTeis.map(tei => buildTei(metadata, program, tei));
+}
+
+export async function getProgram(api: D2Api, id: Id): Promise<Program | undefined> {
+    const {
+        objects: [apiProgram],
+    } = await api.models.programs
+        .get({
+            fields: {
+                id: true,
+                trackedEntityType: { id: true },
+                programTrackedEntityAttributes: {
+                    trackedEntityAttribute: {
+                        id: true,
+                        name: true,
+                        valueType: true,
+                        optionSet: { id: true, options: { id: true, code: true } },
+                    },
+                },
+            },
+            filter: { id: { eq: id } },
+        })
+        .getData();
+
+    if (!apiProgram) return;
+
+    const program: Program = {
+        id: apiProgram.id,
+        trackedEntityType: { id: apiProgram.trackedEntityType.id },
+        attributes: apiProgram.programTrackedEntityAttributes.map(
+            ({ trackedEntityAttribute }) => trackedEntityAttribute
+        ),
+    };
+
+    return program;
+}
+
+type TeiKey = KeysOfUnion<TrackedEntityInstanceApi>;
+
+async function getTeisFromApi(options: {
+    api: D2Api;
+    program: Program;
+    orgUnits: Ref[];
+    page: number;
+    pageSize: number;
+    enrollmentStartDate?: Moment;
+    enrollmentEndDate?: Moment;
+    ouMode: RelationshipOrgUnitFilter;
+}): Promise<PaginatedTeiGetResponse> {
+    const { api, program, orgUnits, page, pageSize, enrollmentStartDate, enrollmentEndDate, ouMode } = options;
+
+    const fields: TeiKey[] = [
+        "trackedEntityInstance",
+        "inactive",
+        "orgUnit",
+        "attributes",
+        "enrollments",
+        "relationships",
+        "featureType",
+        "geometry",
+    ];
+
+    const ouModeQuery =
+        ouMode === "SELECTED" || ouMode === "CHILDREN" || ouMode === "DESCENDANTS"
+            ? { ouMode, ou: orgUnits?.map(({ id }) => id) }
+            : { ouMode };
+
+    return api.trackedEntityInstances
```

Comment:
this is using old tracker api which is deprecated, glass already uses api.tracker.trackedEntities in a lot of places. please reuse that

Reply:
Makes sense, but I'm having some issues converting this endpoint, a lot of the code is reliant on it. Maybe we can make a call and try to address this?

Reply:
@davidbgomes Sure! create a separate PR with the new API and the error it throws ans setup a call with me.